tests/test_operation.py

```diff
@@ -2657,6 +2657,46 @@ def test_mul_does_not_auto_simplify(self):
             assert qml.equal(op[1], op2)
 
 
+@pytest.mark.parametrize(
+    "op",
+    [
+        qml.CZ(wires=[1, 0]),
+        qml.CCZ(wires=[2, 0, 1]),
+        qml.SWAP(wires=[1, 0]),
+        qml.IsingXX(1.23, wires=[1, 0]),
+        qml.Identity(wires=[3, 1, 2, 0]),
+        qml.ISWAP(wires=[1, 0]),
+        qml.SISWAP(wires=[1, 0]),
+        qml.SQISW(wires=[1, 0]),
+        qml.MultiRZ(1.23, wires=[2, 0, 1, 3]),
+        qml.IsingXY(1.23, wires=[1, 0]),
+        qml.IsingYY(1.23, wires=[1, 0]),
+        qml.IsingZZ(1.23, wires=[1, 0]),
+        qml.PSWAP(1.23, wires=[1, 0]),
+    ],
+)
+def test_symmetric_matrix_early_return(op, mocker):
+    """Test that operators that are symmetric over all wires are not reordered
+    when the wire order only contains the same wires as the operator."""
+    if isinstance(op, qml.CZ):
+        pytest.xfail(
+            "CZ is a symmetric op. But, as it is a ControlledOp, "
+            "which overrides Operator.matrix(), this test will fail."
```

Comment:
what do you think of updating `Controlled` to override `compute_matrix` instead of `matrix`? I think it could potentially be done. maybe in another PR, but it would help this

Reply:
Had offline discussion. TL;DR it's not worth the effort to change it for a 2-qubit operator.